File: src/scripts/model/buildings/power/powerConsumer.js

export class PowerConsumer {
    required = 0;
    supplied = 0;

    constructor(required) {
        this.required = required;
        this.supplied = 0
    }

    copy() {
        return new PowerConsumer(this.required)
    }

    isEqual(other) {
        return this.required === other.required && this.supplied === other.supplied;
    }

}

File: Simulation/Model/Buildings/PowerConsumer.cs
namespace Simulation.Model.Buildings;

public class PowerConsumer
{
    public int Required { get; set; }
    public int Supplied { get; set; }

    public PowerConsumer(int required)
    {
        Required = required;
        Supplied = 0;
    }

    public PowerConsumer Copy() => new(Required) { Supplied = Supplied };

    public bool IsEqual(PowerConsumer other) =>
        Required == other.Required && Supplied == other.Supplied;
}
